app/script/storage/StorageRepository.coffee

```diff
@@ -215,45 +215,44 @@ class z.storage.StorageRepository extends cryptobox.CryptoboxStore
     return "#{conversation_id}@#{sender_id}@#{timestamp}"
 
   ###
-  Save an unencrypted conversation event.
+  Save a decrypted conversation event.
 
-  @param event [Object] JSON event to be stored
+  @param primary_key [String] Primary key to save the object with
+  @param otr_message_event [Object] JSON event to be stored
+  @param mapped_json [Object] OTR event mapped to it's decrypted counterpart
   @return [Promise] Promise that resolves with the stored record
   ###
-  save_unencrypted_conversation_event: (event) ->
+  save_decrypted_conversation_event: (primary_key, otr_message_event, mapped_json) ->
     return new Promise (resolve, reject) =>
-      primary_key = @construct_primary_key event.conversation, event.from, event.time
-
       event_object =
-        raw: event
+        raw: otr_message_event
         meta:
-          timestamp: new Date(event.time).getTime()
+          timestamp: new Date(otr_message_event.time).getTime()
           version: 1
+        mapped: mapped_json
+
+      # We don't need to keep ciphertext once it has been successfully decrypted
+      event_object.raw.data = undefined
```

Comment:
Should we change this to some flag instead of undefined so that we know that we successfully decrypted the event?